### PR TITLE
Sort search results

### PR DESCRIPTION
**Pull request by TobyRet**
This PR wires up the sort drop down menu to the back end.
Also provides working functionality for users with no javascript.

### CODE REVIEW

**Comment by TobyRet** [comment]
[“Apply filters” applies sorting as well, it should only apply to filters](https://trello.com/c/snTnmZtb/371-apply-filters-applies-sorting-as-well-it-should-only-apply-to-filters)

File: app/assets/javascripts/find-data.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+function sortDatasets() {
```
Comment (maxf) [inline app/assets/javascripts/find-data.js]: This file's name is confusing. And there are no comments to explain what it does.

File: app/assets/javascripts/find-data.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    document.location.href = newLocation()
+
+    function newLocation() {
```
Comment (maxf) [inline app/assets/javascripts/find-data.js]: I don't think that creating a newLocation function is necessary. It brings the risk of things breaking is the declaration ever changes to ` var newLocation = function...` and stop the hoisting of the function. Better to just write `document.location.href = function () {...`

File: app/assets/javascripts/find-data.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        var currentLocation = document.location.href
+        var sortParam = "&sort="
```
Comment (maxf) [inline app/assets/javascripts/find-data.js]: We use 2 spaces for indentation.

File: app/assets/stylesheets/govuk-overrides.scss [hunk 4]
```diff
@@ -49,3 +49,22 @@
     }
   }
 }
+
+// ______________________________________________________
+
+// Search
+
+.apply-sort {
+  float: right;
```
Comment (maxf) [inline app/assets/stylesheets/govuk-overrides.scss]: This doesn't belong in this file, which is originally intended for modifications to the standard GOV.UK style. This rather belongs to components.scss

File: app/views/search/_sort.html.erb [hunk 5]
```diff
@@ -1,9 +1,14 @@
+<div class='apply-sort hidden'>
+  <input type="submit" value="Apply sorting"/>
```
Comment (maxf) [inline app/views/search/_sort.html.erb]: Did you mean `js-hidden` ?

File: app/views/search/_sort.html.erb [hunk 6]
```diff
@@ -1,9 +1,14 @@
...
+</div>
+
 <div class="dgu-dropdown">
   <label>
     <span class="visuallyhidden">Sorting type</span>
-    <select name="sort">
+    <select id='sort-datasets' name="sort" onchange="sortDatasets()">
       <option <%= 'selected' if @sort!='recent' %> value="best">Best match</option>
```
Comment (maxf) [inline app/views/search/_sort.html.erb]: Why doesn't `onchange="this.form.submit()"` work?

File: app/assets/javascripts/find-data.js (hunk 3 above)
Comment (maxf) [inline app/assets/javascripts/find-data.js]: I recommend (and probably eventually mandate) to run all js files against jslint before committing.

**Pull request merged**